### ISSUE TITLE
Failed to execute 'add' on 'DOMTokenList': The token provided must not be empty

### ISSUE THREAD

**Issue by codeflorist** [nuxt/framework#9057]
### Environment

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0-rc.14`
- Nitro Version:    `1.0.0`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

https://stackblitz.com/edit/nuxt-starter-kn7cz5?file=package.json

### Describe the bug

This app.vue:
```vue
<script lang="ts" setup>
const bodyClass = ref('');
onMounted(() => {
  window.addEventListener('scroll', (e) => {
    bodyClass.value = window.top.scrollY > 100 ? 'fixed-header' : '';
  });
});
</script>

<template>
  <Body :class="bodyClass" />
  <div style="height: 2000px"></div>
</template>
```

...throws the following error in console:
```
index.mjs?v=f97dfb0a:66 Uncaught (in promise) DOMException: Failed to execute 'add' on 'DOMTokenList': The token provided must not be empty.
    at https://nuxt-starter-kn7cz5--3000.local-credentialless.webcontainer.io/_nuxt/node_modules/@unhead/vue/dist/index.mjs?v=f97dfb0a:66:25
    at Array.forEach (<anonymous>)
    at setAttrs (https://nuxt-starter-kn7cz5--3000.local-credentialless.webcontainer.io/_nuxt/node_modules/@unhead/vue/dist/index.mjs?v=f97dfb0a:57:29)
    at renderDOMHead (https://nuxt-starter-kn7cz5--3000.local-credentialless.webcontainer.io/_nuxt/node_modules/@unhead/vue/dist/index.mjs?v=f97dfb0a:142:7)
```

Furthermore, the class `fixed-header` gets correctly set when scrolling down, but does not get removed, when scrolling up again.

The problem was introduced in RC12 and persists up to RC14. RC11 does not throw the error and removes the class correctly.

**Comment by codeflorist** [comment]
it seems, setting `bodyClass` to an empty string is the cause of the problem. so this is a workaround:

```js
bodyClass.value = window.top.scrollY > 100 ? 'fixed-header' : 'something';
```

but what if i do not want a class at all?

**Comment by tntsoft** [comment]
Can confirm it's an issue, can't upgrade to v3.0.0 because of it.

![image](https://user-images.githubusercontent.com/5844176/202630272-059c1222-bf09-4f48-bf70-a8b4e4a4e3d8.png)

It definitely seems to be ```class``` related:

![image](https://user-images.githubusercontent.com/5844176/202630628-9c5631f5-4578-4d2e-b1c8-b4fc228bc446.png)

**Comment by SergisGit** [comment]
I have the same problem with this code:
`
watchEffect(() => {
  useHead({
    htmlAttrs: {
      class: props.modelValue ? "overflow" : ""
    }
  });
});
`

**Comment by Aksoom-Hussain** [comment]
facing same issue even it not empty, work fine in RC11

```ts
useHead({
  bodyAttrs: {
    class: computed(() => {return isNavActive.value ? ' active-navbar-body ':' none '})
  },
})
```

```
Uncaught (in promise) DOMException: Failed to execute 'add' on 'DOMTokenList': The token provided must not be empty.
    at http://localhost:3000/_nuxt/node_modules/.pnpm/@unhead+vue@1.0.0_vue@3.2.45/node_modules/@unhead/vue/dist/index.mjs?v=ff9f31e7:66:25
    at Array.forEach (<anonymous>)

```

**Comment by dissy123** [comment]
I have the same issue when setting body class to an empty String 

`<Body :class="{ 'overflow-hidden': !isMobile && isMobile !== undefined }" >`

`Uncaught (in promise) DOMException: Failed to execute 'add' on 'DOMTokenList': The token provided must not be empty.
    at http://localhost:3000/_nuxt/node_modules/@unhead/vue/dist/index.mjs?v=492a2091:50:25
    at Array.forEach (<anonymous>)
    at setAttrs (http://localhost:3000/_nuxt/node_modules/@unhead/vue/dist/index.mjs?v=492a2091:41:29)
    at renderDOMHead (http://localhost:3000/_nuxt/node_modules/@unhead/vue/dist/index.mjs?v=492a2091:126:7)`



Can confirm that this workaround helps:

`<Body :class="{ 'overflow-hidden': !isMobile && isMobile !== undefined, something: true }" >`


It just appears on Body Element, not on others.

**Comment by manniL** [comment]
cc @harlan-zw

**Comment by harlan-zw** [comment]
Hey everyone, so this is indeed a bug and a fix is vailable in @vueuse/head 1.0.18. 

You can upgrade by deleting lock file / node_modules and re-installing or manually adding this version.

Let me know if you have any issues with that.

Additionally, I'd recommend you switch to the new class API, which would look like this:

```ts
useHead({
  class: {
    'overflow-hidden': isMobile
  }
})
```

```vue
<template>
  <Body :class="{ 'overflow-hidden': isMobile }" />
</template>
```

Similar to the Vue class API. You can also make use of the DOM events API to clean up your code too.

```ts
const isFixedHeader = ref(false)
useHead({
  bodyAttrs: {
    onscroll: (e) => { isFixedHeader.value = window.top.scrollY > 100 },
    class: {
      ['fixed-header']: isFixedHeader
    }
  },
})
```

Please consult the unhead documentation to read more about the new features https://unhead.harlanzw.com/

**Comment by harlan-zw** [comment]
> I have the same problem with this code: `watchEffect(() => { useHead({ htmlAttrs: { class: props.modelValue ? "overflow" : "" } }); });`
> 
> But if you specify a non-empty class, then it is never deleted.

Please avoid wrapping `useHead` with any sort of watcher, just pass in the reactive input.

```ts
useHead({ htmlAttrs: { class: {  overflow: () => props.modelValue } });
```
